src/kaleidoscope/driver/storage/None.h

```diff
@@ -0,0 +1,38 @@
+/* -*- mode: c++ -*-
+ * NoStorage -- Dummy Storage hardware driver for Kaleidoscope
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "kaleidoscope/driver/storage/Base.h"
+
+namespace kaleidoscope {
+namespace driver {
+namespace storage {
+
+struct NoProps : public kaleidoscope::driver::storage::BaseProps {};
```

Comment:
Looking at the code, "NoProps" is misleading. It's not that there are no properties, but that we haven't defined values for whatever properties are there.  Thoughts on other names?

I -almost- feel like it'd be more straightforward to just make the baseclass a concrete no-op version, rather than, essentially, an abstract base class. If you want to override it, great. You get actual behavior. If you don't, no behavior. It'd reduce the amount of code, which is always a win.

Reply:
I'm fine with dropping NoProps, and using `::foo::Base<::foo::BaseProps>` instead.